### PR TITLE
Fix #36

### LINKED ISSUE #36: Error in next_message_reader
The line "8=FIX" will not necessarily be the beginning of a new message. For example, you will get "58=FIX arbitrary string". Search correctly for "\18=FIX"

### PR DESCRIPTION
**Pull request by Roman-Koshelev**
next_message_reader bug fix

### CODE REVIEW

**Comment by jamesdbrock** [comment]
Thank you for the PR! Reviewing...

File: include/hffix.hpp [hunk 1]
```diff
@@ -2068,10 +2068,9 @@ class message_reader {
...
         if (!is_valid_) { // this message isn't valid, so we have to try to search for the beginning of the next message.
             char const* b = buffer_ + 1;
-            while(b < buffer_end_ - 10) {
-                if (!std::memcmp(b, "8=FIX", 5))
+            while(b < buffer_end_ - 11) {
+                if (!std::memcmp(b++, "\18=FIX", 6))
                     break;
-                ++b;
             }
             return message_reader(b, buffer_end_);
```
Comment (jamesdbrock) [inline include/hffix.hpp:2075]: I think maybe this has to be `b+1` now?

**Comment by jamesdbrock** [comment]
I'm not sure about this PR yet. I think in order to merge this I'd want to test it. 

I just added a new section to the README about this: https://github.com/jamesdbrock/hffix#pull-requests

We don't yet have any tests for this finding-next-valid-message feature in the unit test suite. Would you please push another commit which adds a test to the unit test suite which fails because of this bug?

File: include/hffix.hpp (hunk 1 above)
Comment (Roman-Koshelev) [reply include/hffix.hpp:2075]: No)

**Comment by Roman-Koshelev** [comment]
Will do

**Comment by jamesdbrock** [comment]
> Will do

Thanks!

**Comment by Roman-Koshelev** [comment]
Done

**Comment by Roman-Koshelev** [comment]
No, not really. You can't do that

**Pull request closed without merging**